processing/src/main/java/io/druid/query/filter/SelectorDimFilter.java

```diff
@@ -88,10 +88,17 @@ public Filter toFilter()
       return new SelectorFilter(dimension, value);
     } else {
       final String valueOrNull = Strings.emptyToNull(value);
-      final Predicate<String> predicate = new Predicate<String>()
+      final Long valueAsLong = Longs.tryParse(value);
+      final DruidCompositePredicate predicate = new DruidCompositePredicate()
       {
         @Override
-        public boolean apply(String input)
+        public boolean applyLong(long input)
+        {
+          return input == valueAsLong;
```

Comment:
This will NPE if the number was unparseable. Instead, if `valueAsLong` is null, applyLong should return false.